src/main/java/org/isf/medicalinventory/gui/InventoryWardEdit.java

```diff
@@ -467,118 +873,162 @@ private JButton getPrintButton() {
+                    || (dischargeType != null && !dischargeType.getCode().equals(lastDischarge))
+                    || (supplier != null && !supplier.getSupId().equals(lastSupplier))
+                    || (destination == null && lastDestination != null) || (chargeType == null && lastCharge != null)
+                    || (dischargeType == null && lastDischarge != null) || (supplier == null && lastSupplier != null)
+                    || !Objects.equals(lastReference, newReference) || !lastDate.toLocalDate().equals(dateInventory.toLocalDate())
+            ) {
+                int reset = MessageDialog.yesNoCancel(null, "angal.inventory.doyouwanttosavethechanges.msg");
+                if (reset == JOptionPane.YES_OPTION) {
+                    this.saveButton.doClick();
+                }
+                if (reset == JOptionPane.NO_OPTION) {
+                    resetVariable();
+                    dispose();
+                } else {
+                    resetVariable();
+                    return;
+                }
+            } else {
+                resetVariable();
+                dispose();
+            }
         });
         return closeButton;
     }
 
     private JScrollPane getScrollPaneInventory() {
         if (scrollPaneInventory == null) {
             scrollPaneInventory = new JScrollPane();
-            scrollPaneInventory.setViewportView(getJTableInventoryRow());
+            try {
+                scrollPaneInventory.setViewportView(getJTableInventoryRow());
+            } catch (OHServiceException e) {
+                OHServiceExceptionUtil.showMessages(e);
+            }
         }
         return scrollPaneInventory;
     }
 
-    private JTable getJTableInventoryRow() {
+    private JTable getJTableInventoryRow() throws OHServiceException {
         if (jTableInventoryRow == null) {
             jTableInventoryRow = new JTable();
-            jTetFieldEditor = new JTextField();
+            jTextFieldEditor = new JTextField();
             jTableInventoryRow.setFillsViewportHeight(true);
             jTableInventoryRow.setModel(new InventoryRowModel());
+            for (int i = 0; i < pColumnVisible.length; i++) {
+                jTableInventoryRow.getColumnModel().getColumn(i).setCellRenderer(new EnabledTableCellRenderer());
+                jTableInventoryRow.getColumnModel().getColumn(i).setPreferredWidth(pColumwidth[i]);
+                if (i == 0 || !pColumnVisible[i]) {
+                    jTableInventoryRow.getColumnModel().getColumn(i).setMinWidth(0);
+                    jTableInventoryRow.getColumnModel().getColumn(i).setMaxWidth(0);
+                    jTableInventoryRow.getColumnModel().getColumn(i).setPreferredWidth(0);
+                }
+            }
+            DefaultTableCellRenderer centerRenderer = new DefaultTableCellRenderer();
+            centerRenderer.setHorizontalAlignment(JLabel.CENTER);
```

Comment:
```suggestion
            centerRenderer.setHorizontalAlignment(SwingConstants.CENTER);
```
Should use the static base class and not accessed via derived types